### ISSUE TITLE
how to deal with UNKNOWN status

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#92]
In GitLab by @vsicardi on Feb 1, 2016, 12:42

Hi,
During the execution of experiment a034, several times,  I am having UNKNOWN status (White box) at some point (it can be POST, or CLEAN...).
Normally I change the status manually to READY, re-launch and it works. This  implies a constant control of the experiment, but at least it works.
Anyway, today I realized that the UNKNOWN status, can be even a COMPLETED status!
Here a real example.
The UNKNOWN status appeared at the CLEAN of chunck 78. 
I checked the error:
The final lines of a034_19500201_fc0_78_CLEAN.err

++ ls -1 /gpfs/scratch/bsc32/bsc32774/a034/LOG_a034/a034_19500201_fc0_78_CLEAN_1549492.err /gpfs/scratch/bsc32/bsc32774/a034/LOG_a034/a034_19500201_fc0_78_CLEAN_1549492.out /gpfs/scratch/bsc32/bsc32774/a034/LOG_a034/a034_19500201_fc0_78_CLEAN.cmd
+ failed_errfiles=
+ set -e
+ failed_jobs_qt=0
+ failed_jobs_rt=0
+ echo '1454262998 48 237 0 0 0'
+ exit 0

It did not produce "a034_19500201_fc0_78_CLEAN_COMPLETED", but was in fact completed.
In order to have the exp run I did set the status to COMPLETED and create manually a a034_19500201_fc0_78_CLEAN_COMPLETED.
Then re-launch the experiment.

I report it so we have a trace of what happened. Do not know if it can be solved.

thank you
valentina
@obellprat, @ macosta, @dmanubens, @jvegreg

**Comment by kinow** [comment]
In GitLab by @jvegreg on Feb 1, 2016, 15:45

This morning I checked with @obellprat and found that marenostrum's scheduler is returning UNKNOWN when autosubmit checks job status. This behaviour appears after some LSF services go down for a while.

We can not avoid this status from autosubmit side but, as this behaviour is appearing more and more frequently, we need to discuss what strategy to choose when autosubmit get jobs on this status:

- We can check periodically for the COMPLETED file (but not as frequently as with the others jobs)

- We can wait for the WALLCLOCK to be expired and then look for the COMPLETED file

**Comment by kinow** [comment]
In GitLab by @obellprat on Feb 7, 2016, 20:06

Hi Javi,

I think checking for COMPLETED files periodically would be the safest option. This problem slowing my experiments highly down, all experiments get stuck after one day and I'm running 6 in parallel. Would be great if you could give this issue a high priority.

**Comment by kinow** [comment]
In GitLab by @dmanubens on May 13, 2016, 16:20

fixed in 57c509cc061530a4bfd59d689fbdcfdc95af75f9

**Comment by kinow** [comment]
In GitLab by @ jlope2 on May 24, 2016, 17:50

Done! Since the next release, Autosubmit will re-check all the jobs with the UNKNOWN status.

Thanks!

**Comment by kinow** [comment]
In GitLab by @fmassonn on Jun 23, 2016, 09:25

Hi,

Related to issue #147 , I have had a similar problem:

```
fmassonn@ bscearth321: ~/logs/autosubmit >> cat /home/Earth/fmassonn/logs/autosubmit/log_a07m_20160622135708_bscearth321 | tail -20


5389 of 11666 jobs remaining (21:43)
[WARNING] Job a07m_19790101_fc22_65_POST seemed to be completed but there is no COMPLETED file
[WARNING] Job a07m_19790101_fc22_65_POST in UNKNOWN status
Job a07m_19790101_fc23_65_POST is RUNNING
Job a07m_19790101_fc24_65_POST is RUNNING


5389 of 11666 jobs remaining (21:45)
kill: sending signal to 1956728 failed: No such process
Job a07m_19790101_fc22_65_POST is COMPLETED
Job a07m_19790101_fc23_65_POST is COMPLETED
Job a07m_19790101_fc24_65_POST is RUNNING


5387 of 11666 jobs remaining (21:46)
Job a07m_19790101_fc24_65_POST is COMPLETED
No more jobs to run.
Some jobs have failed and reached maximun retrials
```

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 4, 2018, 11:50

This seem solved in #147 two years ago so I'm not sure if this is happening anymore. If this issue is still in the latest version please, open a new issue with updated information.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Dec 4, 2018, 11:50

closed
